sos/sosreport.py

```diff
@@ -367,9 +367,51 @@ def __init__(self, args):
             self.tempfile_util.clean()
             self._exit(1)
 
+        self._get_hardware_devices()
+
     def print_header(self):
         print("\n%s\n" % _("sosreport (version %s)" % (__version__,)))
 
+    def _get_hardware_devices(self):
+        self.devices = {
+            'block': self.get_block_devs(),
+            'fibre': self.get_fibre_devs()
+        }
+        # TODO: enumerate network devices, preferably with devtype info
+
+    def get_fibre_devs(self):
+        '''Enumerate a list of fibrechannel devices on this system so that
+        plugins can iterate over them
+
+        These devices are used by add_fibredev_cmd() in the Plugin class.
+        '''
+        try:
+            devs = []
+            devdirs = [
+                'fc_host',
+                'fc_transport',
+                'fc_remote_ports',
+                'fc_vports'
+            ]
+            for devdir in devdirs:
+                if os.isdir(devdir):
```

Comment:
Question from devices noob: is this test correct? To check for `./fc_host` directory presence (and others in `pwd`)?

Shall not we test `os.isdir("/sys/class/%s" % devdir)` or similar?

Reply:
Good catch, looks like I inadvertently dropped that when refactoring the discovery bits. I'll update that later today.